crates/iceberg/src/io.rs

```diff
@@ -206,6 +205,35 @@ impl FileIO {
     }
 }
 
+/// The struct the represents the metadata of a file.
+///
+/// TODO: we can add last modified time, content type, etc. in the future.
+pub struct FileMetadata {
+    /// The size of the file.
+    pub size: u64,
+}
+
+/// Trait for reading file.
+///
+/// # TODO
+///
+/// It's possible for us to remove the async_trait, but we need to figure
+/// out how to handle the object safety.
+#[async_trait::async_trait]
+pub trait FileRead: Send + Unpin + 'static {
```

Comment:
Maybe we could call this `IcebergFileRead` or `IcebergRead`? I suppose that's a bit redundant as it would be clear that it is from us by navigating to the `use` statement where it gets imported, but I'm just conscious that there are a lot of `Read` and `Write` traits dotted around and we could make it easier to see which one is ours at-a-glance where ever it is used in the future.

Reply:
I'm open to either name. This PR aims to minimize changes to our existing traits, so I prefer moving this discussion to a separate issue. What do you think?

Reply:
Ok - fine with me.

Reply:
Tracked at https://github.com/apache/iceberg-rust/issues/368